src/GraphQl/Resolver/FieldsToAttributesTrait.php

```diff
@@ -34,15 +34,17 @@ private function fieldsToAttributes(ResolveInfo $info): array
 
     private function replaceIdKeys(array $fields): array
     {
+        $denormalizedFields = [];
         foreach ($fields as $key => $value) {
             if ('_id' === $key) {
-                $fields['id'] = $fields['_id'];
-                unset($fields['_id']);
-            } elseif (\is_array($fields[$key])) {
-                $fields[$key] = $this->replaceIdKeys($fields[$key]);
+                $denormalizedFields['id'] = $fields['_id'];
+                continue;
             }
+            $denormalizedFields[$this->denormalizePropertyName($key)] = \is_array($fields[$key]) ? $this->replaceIdKeys($fields[$key]) : $value;
         }
 
-        return $fields;
+        return $denormalizedFields;
     }
+
+    abstract protected function denormalizePropertyName($property);
```

Comment:
Why not be stricter?

```suggestion
    abstract protected function denormalizePropertyName(string $property): string;
```

Reply:
The idea was both to respect the underlying Symfony NameConverterInterface and to be in line with what has been done in ORM filters ([see original discussion](https://github.com/api-platform/core/pull/2751#discussion_r277631577))
I'm 100% ok to update if those args are not worth it, let me know

Reply:
Personally, I would prefer a stricter approach. I think Symfony will change its interface soon too.

Reply:
@alanpoulain is right the interfaces will be changed in 5.0